it/services/EnsureGuaranteeServiceIntegrationSpec.scala

```diff
@@ -7,77 +7,184 @@ import org.scalatest.concurrent.{IntegrationPatience, ScalaFutures}
+
+        normalise(result.right.get) mustEqual normalise(TestData.buildGBEUXml(TestData.otherInputXML))
+      }
+      "must add default special mentions to first good if no special mention present for guarantees" in {
+        val result = service.ensureGuarantee(TestData.buildGBEUXml(TestData.extraGuaranteesInputXML))
+
+        normalise(result.right.get) mustEqual normalise(TestData.buildGBEUXml(TestData.extraGuaranteesExpectedXML))
+        validator.validate(normalise(result.right.get), DepartureDeclarationXSD) mustBe a[Right[_, XmlValid]]
+      }
+      "must add default special mentions to first good if special mention isn't present for a guarantee" in {
+        val result = service.ensureGuarantee(TestData.buildGBEUXml(TestData.extraGuaranteesComboInputXML))
+
+        normalise(result.right.get) mustEqual normalise(TestData.buildGBEUXml(TestData.extraGuaranteesComboExpectedXML))
+        validator.validate(normalise(result.right.get), DepartureDeclarationXSD) mustBe a[Right[_, XmlValid]]
+      }
+      "must allow non-guarantee special mentions through without issue" in {
+        val result = service.ensureGuarantee(TestData.buildGBEUXml(TestData.oddSpecialMentionsInputXml))
+
+        result.right.get.toString().filter(_ > ' ') mustEqual TestData.buildGBEUXml(TestData.oddSpecialMentionsOutputXml).toString().filter(_ > ' ')
+        validator.validate(result.right.get.toString().filter(_ > ' '), DepartureDeclarationXSD) mustBe a[Right[_, XmlValid]]
+
+      }
+      "must allow CAL special mentions with additional values through without removing fields" in {
+        val result = service.ensureGuarantee(TestData.buildGBEUXml(TestData.mixedSpecialMentionsInputXml))
+
+        normalise(result.right.get) mustEqual normalise(TestData.buildGBEUXml(TestData.mixedSpecialMentionsOutputXml))
+        validator.validate(normalise(result.right.get), DepartureDeclarationXSD) mustBe a[Right[_, XmlValid]]
+      }
     }
+  }
 
-    "result must pass standard validation" in {
-      val service = app.injector.instanceOf[EnsureGuaranteeService]
-      val validator = app.injector.instanceOf[XmlValidationService]
-
-      val result = service.ensureGuarantee(TestData.buildXml(TestData.standardInputXML))
-
-      validator.validate(result.right.get.toString().filter(_ > ' '), DepartureDeclarationXSD) mustBe a[Right[_, XmlValid]]
+  "listed scenarios" - {
+    "GB -> EU, type 1, CAL w/o Reference Number - SM Passes through to core unedited, new one added" in {
+      val insertXml = TestData.basicGuarantee ++ TestData.goodsWithCustomSpecialMention(<SPEMENMT2><AddInfCodMT23>CAL</AddInfCodMT23></SPEMENMT2>)
+      val xml = TestData.buildGBEUXml(insertXml)
+      val expectedXml = TestData.buildGBEUXml(TestData.basicGuarantee ++ TestData.goodsWithCustomSpecialMention(<SPEMENMT2><AddInfCodMT23>CAL</AddInfCodMT23></SPEMENMT2><SPEMENMT2><AddInfMT21>1.00GBP21GB0000010000HU1</AddInfMT21><AddInfCodMT23>CAL</AddInfCodMT23></SPEMENMT2>))
```

Comment:
Could we possibly extract some of these large XML nodes into their own `val`s and format them so they aren't on one line?

Reply:
Will do